pkg/provider/aws/execution_unit.go

```diff
@@ -107,21 +107,28 @@ func (a *AWS) GenerateExecUnitResources(unit *core.ExecutionUnit, result *core.C
 							if val.ExecUnitName != unit.ID {
 								continue
 							}
-							switch val.Type {
-							// TODO handle kubernetes.TargetGroupTransformation
-							case string(kubernetes.ImageTransformation):
+							switch kubernetes.ProviderValueTypes(val.Type) {
+							case kubernetes.ImageTransformation:
 								khChart.Values[val.Key] = core.IaCValue{
 									Resource: image,
 									Property: resources.ECR_IMAGE_NAME_IAC_VALUE,
 								}
 								dag.AddDependency(khChart, image)
-							case string(kubernetes.ServiceAccountAnnotationTransformation):
+							case kubernetes.ServiceAccountAnnotationTransformation:
 								khChart.Values[val.Key] = core.IaCValue{
 									Resource: role,
 									Property: core.ARN_IAC_VALUE,
 								}
 								dag.AddDependency(khChart, role)
-							case string(kubernetes.TargetGroupTransformation):
+							case kubernetes.InstanceTypeKey:
+								khChart.Values[val.Key] = core.IaCValue{
+									Property: "eks.amazonaws.com/nodegroup",
+								}
+							case kubernetes.InstanceTypeValue:
+								khChart.Values[val.Key] = core.IaCValue{
+									Property: resources.NodeGroupNameFromConfig(cfg),
+								}
```

Comment:
wouldnt this only be conditionally set? If instance type and disk size arent set on the config then we wouldnt want to place it on a node group. Same with the key

Reply:
The Key and Value are only conditionally set in the provider values, so that is handled here: https://github.com/klothoplatform/klotho/pull/487/files#diff-9da19c3bee3a0431ac52dba95b3211cfd88e3a40eb4f484064ddf771f0b0a2abR166

Reply:
ah good point 👍